signature-aggregator/aggregator/aggregator.go

```diff
@@ -287,87 +288,42 @@ func (s *SignatureAggregator) getExcludedValidators(
 	return excludedValidators, nil
 }
 
-func (s *SignatureAggregator) CreateSignedMessage(
+func validateQuorumPercentages(required, buffer uint64) error {
+	if required == 0 || required+buffer > 100 {
+		return errInvalidQuorumPercentage
+	}
+	return nil
+}
+
+func (s *SignatureAggregator) selectSigningSubnet(
 	ctx context.Context,
 	log logging.Logger,
 	unsignedMessage *avalancheWarp.UnsignedMessage,
-	justification []byte,
 	inputSigningSubnet ids.ID,
-	requiredQuorumPercentage uint64,
-	quorumPercentageBuffer uint64,
-	skipCache bool,
-) (*avalancheWarp.Message, error) {
-	if requiredQuorumPercentage == 0 || requiredQuorumPercentage+quorumPercentageBuffer > 100 {
-		log.Error(
-			"Invalid quorum percentages",
-			zap.Uint64("requiredQuorumPercentage", requiredQuorumPercentage),
-			zap.Uint64("quorumPercentageBuffer", quorumPercentageBuffer),
-		)
-		return nil, errInvalidQuorumPercentage
-	}
-
-	log.Debug("Creating signed message")
-	var signingSubnet ids.ID
-	var err error
-	// If signingSubnet is not set we default to the subnet of the source blockchain
-	sourceSubnet, err := s.getSubnetID(ctx, log, unsignedMessage.SourceChainID)
+) (ids.ID, ids.ID, error) {
+	sourceSubnetID, err := s.getSubnetID(ctx, log, unsignedMessage.SourceChainID)
 	if err != nil {
-		return nil, fmt.Errorf(
+		return ids.ID{}, ids.ID{}, fmt.Errorf(
 			"source message subnet not found for chainID %s",
 			unsignedMessage.SourceChainID,
 		)
 	}
+
+	var signingSubnetID ids.ID
 	if inputSigningSubnet == ids.Empty {
-		signingSubnet = sourceSubnet
+		signingSubnetID = sourceSubnetID
 	} else {
-		signingSubnet = inputSigningSubnet
-	}
-	log.Debug(
-		"Creating signed message with signing subnet",
-		zap.Stringer("signingSubnet", signingSubnet),
-	)
-
-	connectedValidators, err := s.connectToQuorumValidators(ctx, log, signingSubnet, requiredQuorumPercentage, skipCache)
-	if err != nil {
-		log.Error(
-			"Failed to fetch quorum of connected canonical validators",
-			zap.Stringer("signingSubnet", signingSubnet),
-			zap.Error(err),
-		)
-		return nil, err
-	}
-
-	isL1 := false
-	if signingSubnet != constants.PrimaryNetworkID {
-		isL1, err = s.isSubnetL1(ctx, log, signingSubnet)
-		if err != nil {
-			return nil, fmt.Errorf("failed to check if signing subnet is L1: %w", err)
-		}
+		signingSubnetID = inputSigningSubnet
 	}
+	return signingSubnetID, sourceSubnetID, nil
+}
 
-	// Tracks all collected signatures.
-	// For L1s, we must take care to *not* include inactive validators in the signature map.
-	// Inactive validator's stake weight still contributes to the total weight, but the verifying
-	// node will not be able to verify the aggregate signature if it includes an inactive validator.
+func (s *SignatureAggregator) populateSignatureMapFromCache(
```

Comment:
```suggestion
// Gets all of the signatures for the given message that have been cached from the connected valdiators.
// Excludes previously fetched signatures from any validators now inactive.
// Returns the valid cached signatures to be used, and the total weight of the validators those signatures represent.
func (s *SignatureAggregator) getCachedSignaturesForMessage(
```

Naming/documentation nit.